test/test-PerfMeter.js

```diff
@@ -0,0 +1,162 @@
+/* globals Services */
+
+"use strict";
+const simplePrefs = require("sdk/simple-prefs");
+
+const {before, after} = require("sdk/test/utils");
+const tabs = require("sdk/tabs");
+const {setTimeout} = require("sdk/timers");
+const {ActivityStreams} = require("lib/ActivityStreams");
+const {Cu} = require("chrome");
+Cu.import("resource://gre/modules/Services.jsm");
+
+let ACTIVITY_STREAMS_URL;
+let app;
+
+function openTestTab(notifyEvent) {
+  let tabData = {};
+  tabData.showPromise = new Promise(resolve => {
+    function onShow(tab) {
+      tabData.tab = tab;
+      tabs.removeListener("pageshow", onShow);
+      if (!notifyEvent) {
+        setTimeout(function() {
```

Comment:
You want all other observers of "pageshow" event to handle that event before you resolve the promise.  Which is why you stick resolve inside the timeout, so the promise is resolved after everyone else finish with  "pageshow"  event.